skbio/alignment/_alignment.py

```
@@ -72,13 +72,16 @@ def __init__(self, seqs):
         self._data = seqs
         self._id_to_index = {}
         for i, seq in enumerate(self._data):
-            id = seq.id
-            if id in self:
+            if 'id' not in seq.metadata:
```

should raise an error if id is missing